src/common/utils.ts

```diff
@@ -279,3 +279,29 @@ export const getCurrencySymbol = ({
 
 export const concatTextToNewline = (...args: Array<string | undefined>) =>
   args.filter(Boolean).join(`\n`);
+
+export function snakeToCamel<S extends string>(str: S): CamelCasedKey<S> {
+  return str.replace(/_([a-z])/g, (_, letter) =>
+    letter.toUpperCase(),
+  ) as CamelCasedKey<S>;
+}
+
+type CamelCasedKey<S extends string> = S extends `${infer Head}_${infer Tail}`
+  ? `${Head}${Capitalize<CamelCasedKey<Tail>>}`
+  : S;
+
+export type ObjectSnakeToCamelCase<T extends object> = {
+  [K in keyof T as K extends string ? CamelCasedKey<K> : never]: T[K];
+};
+
+export function convertObjectKeysToCamelCase<T extends object>(
```

Comment:
Can we do this statically, we know which properties map to which, also this kind of snake to camel would be for free with protobuf. So basically just map it by hand.

Reply:
Kept the type magic to make it easier for types. Moved to manual mapping now.

Reply:
would it not be more consistent to return camel cased from SkadiApiClient and then everywhere else you use camel cased? Then you have to do the mapping only once inside the client.

Reply:
Then, if we only want the mappings from the client, should we not do the generic function since we are pretty sure that we just want to have them converted?

Reply:
Totally agree, we should phase it out. Just to get started with production testing. After that, I'll start cleaning it up.